modules/generator/generator_kafka.go

```diff
@@ -47,65 +44,13 @@ func (g *Generator) listenKafka(ctx context.Context) {
 				level.Error(g.logger).Log("msg", "readKafka failed", "err", err)
 				continue
 			}
-		case <-ctx.Done():
-			return
 		}
 	}
 }
 
 func (g *Generator) readKafka(ctx context.Context) error {
-	fallback := time.Now().Add(-time.Minute)
-
-	groupLag, err := getGroupLag(
-		ctx,
-		kadm.NewClient(g.kafkaClient),
-		g.cfg.Ingest.Kafka.Topic,
-		g.cfg.Ingest.Kafka.ConsumerGroup,
-		fallback,
-	)
-	if err != nil {
-		return fmt.Errorf("failed to get group lag: %w", err)
-	}
-
-	assignedPartitions := g.getAssignedActivePartitions()
-
-	for _, partition := range assignedPartitions {
-		if ctx.Err() != nil {
-			return ctx.Err()
-		}
-
-		partitionLag, ok := groupLag.Lookup(g.cfg.Ingest.Kafka.Topic, partition)
-		if !ok {
-			return fmt.Errorf("lag for partition %d not found", partition)
-		}
-
-		if partitionLag.Lag <= 0 {
-			// Nothing to consume
-			continue
-		}
-
-		err := g.consumePartition(ctx, partition, partitionLag)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func (g *Generator) consumePartition(ctx context.Context, partition int32, lag kadm.GroupMemberLag) error {
 	d := ingest.NewDecoder()
 
-	// We always rewind the partition's offset to the commit offset by reassigning the partition to the client (this triggers partition assignment).
-	// This is so the cycle started exactly at the commit offset, and not at what was (potentially over-) consumed previously.
-	// In the end, we remove the partition from the client (refer to the defer below) to guarantee the client always consumes
-	// from one partition at a time. I.e. when this partition is consumed, we start consuming the next one.
-	g.kafkaClient.AddConsumePartitions(map[string]map[int32]kgo.Offset{
```

Comment:
Can you help clarify for my understanding - since this client update is removed, how does PollFetches know which partitions are assigned?  Is that now handled by the server (broker, etc)?

Reply:
Yes, when joining a group (ie. `kgo.ConsumerGroup(kafkaCfg.ConsumerGroup)`) Kafka handles partitions assignment. The client will talk to the broker which will share the partition balance for them. In each pull, the client will receive fetches for the partitions that have been assigned to it.